Game2048Base/src/main/java/models/square/EmptySquare.java

```diff
@@ -0,0 +1,64 @@
+package models.square;
+
+
+import javalib.worldimages.OutlineMode;
+import javalib.worldimages.OverlayImage;
+import javalib.worldimages.Posn;
+import javalib.worldimages.RectangleImage;
+
+import java.awt.Color;
+
+public class EmptySquare extends Square {
+
+    public EmptySquare() {
+        this.value = 0;
+        this.image =
+                new OverlayImage
+                        (new RectangleImage(Square.SIDE_LENGTH - 5,
+                                Square.SIDE_LENGTH - 5,
+                                OutlineMode.SOLID,
+                                Color.GRAY),
+                                (new RectangleImage(Square.SIDE_LENGTH,
+                                        Square.SIDE_LENGTH,
+                                        OutlineMode.SOLID,
+                                        Color.DARK_GRAY)));
+    }
+
+    public EmptySquare(Posn position) {
+        this.position = position;
+        this.value = 0;
+        this.image =
+                new OverlayImage
+                        (new RectangleImage(Square.SIDE_LENGTH - 5,
+                                Square.SIDE_LENGTH - 5,
+                                OutlineMode.SOLID,
+                                Color.GRAY),
+                                (new RectangleImage(Square.SIDE_LENGTH,
+                                        Square.SIDE_LENGTH,
+                                        OutlineMode.SOLID,
+                                        Color.DARK_GRAY)));
+    }
+
+    @Override
+    public boolean isTile() { return false; }
+
+    @Override
+    public boolean isEmptyTile() {
```

Comment:
These interface elements seem pretty weird to me. They definitely feel academic in that they look like something you'd learn in an intro to java class. They don't really do much other than provide you a way to avoid instanceof I guess. I think we're too late into this development process for me to understand what to do instead because there's so much code already.